homeassistant/components/ecobee/sensor.py

```diff
@@ -111,6 +117,17 @@ def device_info(self) -> DeviceInfo | None:
             )
         return None
 
+    @property
+    def native_unit_of_measurement(self):
```

Comment:
I think it would be better to update `self._attr_native_unit_of_measurement` inside the `__init__` method.
Then you can drop the property.

Reply:
I'm not sure, the other classes here use the property func so I feel that would be inconsistent.

Reply:
I insist. You shouldn't create new properties if you can avoid it.
Some of the other properties would also benefit from refactoring (in a separate PR)

Reply:
That makes sense, although I tried this and inside of `__init__` it throws an error that `self.hass` is `NoneType` and the `hass.config` can not be found.

Reply:
Can you do it in `added_to_hass`?